pkg/util/containers/providers/windows/provider.go

```diff
@@ -0,0 +1,83 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017-2020 Datadog, Inmetrics.
+
+// +build windows
+
+package windows
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/metrics"
+)
+
+// Provider is a Windows implementation of the ContainerImplementation interface
+type Provider struct {
+}
+
+// Prefetch gets data from all cgroups in one go
+// If not successful all other calls will fail
+func (mp *Provider) Prefetch() error {
```

Comment:
It's called each time an access to a method listing ALL containers is done. To avoid doing API calls in the GetXxx() methods. Though it's not mandatory to implement it.